Packs/XSOAR_Engineer_Training/Integrations/XSOAREngineerTraining/XSOAREngineerTraining.py

```diff
@@ -0,0 +1,369 @@
+            'EntryContext': {
+                'ActiveDirectory.Users(obj.dn == val.dn)': user_output.get('attributes'),
+                'Account(obj.ID == val.ID)': user_output.get('account')
+            }
+        }
+
+        return demisto_entry
+    else:
+        return "No user found"
+
+
+def siem_search_command(args):
+    """
+    Returns the results from the siem-search command
+    """
+    query = args.get('query')
+    result_type = args.get('result_type')
+    number = randint(1, 10)
+
+    # return results for demo of siem search in example playbook.
+    if result_type == "email" or query.startswith("email"):
+        result = [
+            {
+                "Username": "XSOAR007",
+                "Email": "james.bond@xsoar.local"
+            },
+            {
+                "Username": "XSOAR001",
+                "Email": "m@xsoar.local"
+            }
+        ]
+    # return some data if the number is right.
+    elif number >= 7 or result_type == "hosts" or query.startswith("username"):
+        result = [
+            {
+                "Host": "crossiscoming01",
+                "Online": "Yes"
+            },
+            {
+                "Host": "crossiscoming02",
+                "Online": "No"
+            }
+        ]
+    # return some data if the number is right.
+    elif query.startswith("ip"):
+        result = [
+            {
+                "Host": "crossiscoming01",
+                "Online": "Yes"
+            }
+        ]
+    else:
+        result = []
+
+    results = CommandResults(
+        readable_output=tableToMarkdown(f"SIEM Search results for query: {query}", result),
+        outputs_prefix="SIEM.Result",
+        outputs=result)
+
+    return results
+
+
+def fetch_incidents(limit):
+    """
+    fetch Incidents
+    """
+    incidents = []
+
+    count = randrange(10, limit)
+    events = mock_data(count)
+
+    for event in events:
+        if event.get("type") == "url allowed":
+            event_type = "URL Allowed"
+        else:
+            event_type = "URL Blocked"
+        event_user = event.get("sourceUser", "")
+        incident = {
+            "name": f"Alert - {event_type}- {event_user}",
+            "rawJSON": json.dumps(event),
+            "occurred": event["occurred"]
+        }
+        incidents.append(incident)
+
+    # return our list of Incidents
+    return incidents[:limit]
+
+
+''' COMMANDS MANAGER / SWITCH PANEL '''
+
+
+def main():
+    """
+    main function to run things
+    """
+    fetch_limit = 20
+
+    command = demisto.command()
+    LOG(f'Command being called is {command}')
+
+    try:
+        commands = {
+            'xsoar-engineer-get-events': get_events_command,
+            'ad-expire-password': simple_response_command,
+            'ad-set-new-password': simple_response_command,
+            'send-mail': simple_response_command
+        }
```

Comment:
Please change the names to all start with the same prefix, for example 'training-ad-set-new-password' or something similar.

Please make the changes to all relevant places, readme .yml etc.

Reply:
Done, updated and added xet- to all commands, short for XSOAR Engineer Training